lib/request.js

```diff
@@ -1,18 +1,28 @@
 const http = require('http');
 const https = require('https');
+const zlib = require('zlib');
 
 module.exports = (options) => {
     options = Object.assign({
         protocol: 'https:'
     }, options);
     const httpModule = options.protocol === 'https:' ? https : http;
     return new Promise((resolve, reject) => {
-        const chunks = [];
         const request = httpModule.request(options, (response) => {
-            response.on('data', (chunk) => {
+            let bodyStream;
+            const chunks = [];
+            const encoding = response.headers && response.headers['content-encoding'];
+            if (encoding === 'gzip' || encoding === 'deflate') {
+                response.on('error', reject);
+                bodyStream = response.pipe(zlib.createUnzip());
```

Comment:
It's handled in `bodyStream.on('error', ...)` because `pipe` returns its destination stream. Also, it's tested in the test case at the bottom.